src/runner/index.js

```diff
@@ -51,6 +50,19 @@ export default class Runner extends EventEmitter {
         return reporter.testCount - reporter.passed;
     }
 
+    _runTask (reporterPlugin, browserSet, tests) {
+        var task     = new Task(tests, browserSet.connections, this.proxy, this.opts);
+        var reporter = new Reporter(reporterPlugin, task, this.opts.reportOutStream);
+
+        task.on('browser-job-done', job => browserSet.releaseConnection(job.browserConnection));
+
+        var promise = this._waitTaskResult(task, browserSet, reporter);
+
+        promise.cancel = async () => await Runner._disposeTaskAndBrowsers(task, browserSet);
```

Comment:
I wonder if it can cause problems if we `cancel` after run promise gets fulfilled?